File: app/Models/Service.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Service extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'business_id',
        'nombre',
        'descripcion',
        'precio',
        'duracion_minutos',
        'buffer_pre_minutos',
        'buffer_post_minutos',
        'requiere_confirmacion',
        'activo',
        'meta',
    ];

    protected $casts = [
        'precio' => 'decimal:2',
        'duracion_minutos' => 'integer',
        'buffer_pre_minutos' => 'integer',
        'buffer_post_minutos' => 'integer',
        'requiere_confirmacion' => 'boolean',
        'activo' => 'boolean',
        'meta' => 'array',
    ];

    /**
     * Global Scope Multi-Tenant: Filtrar por business_id del usuario autenticado
     */
    protected static function booted(): void
    {
        static::addGlobalScope('business', function (Builder $builder) {
            if (auth()->check() && auth()->user()->current_business_id) {
                $builder->where('business_id', auth()->user()->current_business_id);
            }
        });
    }

    /**
     * Relación: Servicio pertenece a un Negocio
     */
    public function business(): BelongsTo
    {
        return $this->belongsTo(Business::class);
    }

    /**
     * Relación: Servicio puede ser realizado por muchos Empleados (N:M)
     */
    public function employees(): BelongsToMany
    {
        return $this->belongsToMany(Employee::class, 'employee_services');
    }
}

File: mobile/lib/data/models/service.dart
class Service {
  final int id;
  final int businessId;
  final String nombre;
  final String? descripcion;
  final int duracionMinutos;
  final double precio;
  final int bufferPreMinutos;
  final int bufferPostMinutos;
  final bool activo;
  final Map<String, dynamic>? meta;

  Service({
    required this.id,
    required this.businessId,
    required this.nombre,
    this.descripcion,
    required this.duracionMinutos,
    required this.precio,
    this.bufferPreMinutos = 0,
    this.bufferPostMinutos = 0,
    required this.activo,
    this.meta,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    final activoRaw = json['activo'];

    return Service(
      id: (json['id'] as num?)?.toInt() ?? 0,
      businessId: (json['business_id'] as num?)?.toInt() ?? 0,
      nombre: (json['nombre'] ?? 'Servicio') as String,
      descripcion: json['descripcion'] as String?,
      duracionMinutos: (json['duracion_minutos'] as num?)?.toInt() ?? 0,
      precio: (json['precio'] as num?)?.toDouble() ?? 0,
      bufferPreMinutos: (json['buffer_pre_minutos'] as num?)?.toInt() ?? 0,
      bufferPostMinutos: (json['buffer_post_minutos'] as num?)?.toInt() ?? 0,
      activo: activoRaw is bool
          ? activoRaw
          : ((activoRaw as num?)?.toInt() ?? 1) == 1,
      meta: json['meta'] as Map<String, dynamic>?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'business_id': businessId,
      'nombre': nombre,
      'descripcion': descripcion,
      'duracion_minutos': duracionMinutos,
      'precio': precio,
      'buffer_pre_minutos': bufferPreMinutos,
      'buffer_post_minutos': bufferPostMinutos,
      'activo': activo,
      'meta': meta,
    };
  }

  String get precioFormateado => '\$${precio.toStringAsFixed(2)}';
  String get duracionFormateada => '$duracionMinutos min';
}
